### PR TITLE
CI, MAINT: switch to stable python release

### LINKED ISSUE #19740: CI, MAINT: some easy cleanups for Python version
Just glancing at the CI config code flanking the diffs in a few recent PRs, you can see some places where switching to the stable release makes sense: `git grep -E -i -P "3\.\d[2]-dev"`

```
.github/workflows/linux_meson.yml:        python-version: ['3.10', '3.12-dev']
.github/workflows/linux_meson.yml:            python-version: '3.12-dev'
.github/workflows/linux_meson.yml:      if: matrix.python-version == '3.12-dev'
```

### PR DESCRIPTION
**Pull request by willtryagain**
#### Reference issue
closes #19740

#### What does this implement/fix?
switch to stable release

### CODE REVIEW

File: .github/workflows/linux_meson.yml [hunk 1]
```diff
@@ -36,12 +36,12 @@ jobs:
     runs-on: ubuntu-22.04
     strategy:
       matrix:
-        python-version: ['3.10', '3.12-dev']
+        python-version: ['3.10', '3.12'] # these are the runs which would use python dev versions
```
Comment (lucascolley) [inline .github/workflows/linux_meson.yml:39]:
```suggestion
        python-version: ['3.10', '3.12'] # this run will use python dev versions when available
```

File: .github/workflows/linux_meson.yml [hunk 2]
```diff
@@ -67,7 +67,7 @@ jobs:
         python -m pip install numpy cython pytest pytest-xdist pytest-timeout pybind11 mpmath gmpy2 pythran ninja meson click rich-click doit pydevtool pooch hypothesis
 
     - name: Install Python packages from repositories
-      if: matrix.python-version == '3.12-dev'
+      if: matrix.python-version == '3.12' # these are the runs which would use python dev versions
```
Comment (lucascolley) [inline .github/workflows/linux_meson.yml:70]: likewise here with rephrasing the comment

**Review by lucascolley — COMMENTED** [review]
could we rephrase the comment like this? Otherwise I think this looks good.

File: .github/workflows/linux_meson.yml (hunk 2 above)
Comment (willtryagain) [reply .github/workflows/linux_meson.yml]: yes, I fixed the grammar

**Review by lucascolley — APPROVED** [review]
looks good to me, thanks @willtryagain ! Unless you'd prefer for the comments to be somewhere else @andyfaff ?

**Review by tylerjereddy — APPROVED** [review]
LGTM as well, I hit the button to flush the GitHub actions CI just in case.

**Comment by tylerjereddy** [comment]
thanks for your first contribution to SciPy

**Comment by willtryagain** [comment]
thanks. it was fun 😀

**Pull request merged**